pkg/cmd/grafana-cli/commands/scuemata_validation_command.go

```diff
@@ -11,18 +13,53 @@ import (
 var paths = load.GetDefaultLoadPaths()
 
 func (cmd Command) validateScuemataBasics(c utils.CommandLine) error {
-	if err := validate(paths, load.BaseDashboardFamily); err != nil {
+	if err := validateScuemata(paths, load.BaseDashboardFamily); err != nil {
 		return err
 	}
 
-	if err := validate(paths, load.DistDashboardFamily); err != nil {
+	if err := validateScuemata(paths, load.DistDashboardFamily); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func validate(p load.BaseLoadPaths, loader func(p load.BaseLoadPaths) (schema.VersionedCueSchema, error)) error {
+func (cmd Command) validateResources(c utils.CommandLine) error {
+	resource := c.String("dashboard")
+	b, err := os.Open(filepath.Clean(resource))
+	if err != nil {
+		return err
+	}
+
+	if err := validateResources(b, paths, load.BaseDashboardFamily); err != nil {
+		return err
+	}
+
+	if err := validateResources(b, paths, load.DistDashboardFamily); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateResources(resource interface{}, p load.BaseLoadPaths, loader func(p load.BaseLoadPaths) (schema.VersionedCueSchema, error)) error {
+	dash, err := loader(p)
+	if err != nil {
+		return fmt.Errorf("error while loading dashboard scuemata, err: %w", err)
+	}
+
+	// Validate checks that the resource is correct with respect to the schema.
+	if resource != nil {
+		err = dash.Validate(schema.Resource{Value: resource})
+		if err != nil {
+			return fmt.Errorf("invalid resource with respect to the schema, err: %w", err)
```

Comment:
Let's simplify this string to just "Failed validation:", then try to print one validation failure per line. (Can do the one-per-line after doing #33934 .)

Reply:
Oh wait, right, that one's merged now. OK, so catch this PR up to that and change this to multi-line-friendliness.

Reply:
Can change to `Failed validation:` but the wrapper is not applied here, since the produced error is of type `core/adt.ValueError`. Maybe we can add it as well to the wrapper, but encountered the same error when I was trying to actually validate produce JSON files from CUE ones (sanity checks) so I can show some love to the wrapper in a follow-up PR?

Reply:
SGTM